File: app/src/main/java/com/kabirnayeem99/paymentpaid/adapters/PaymentAdapter.java
package com.kabirnayeem99.paymentpaid.adapters;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kabirnayeem99.paymentpaid.R;
import com.kabirnayeem99.paymentpaid.utils.CustomUtils;

import java.util.ArrayList;
import java.util.List;

public class PaymentAdapter extends RecyclerView.Adapter<PaymentAdapter.ViewHolder> {
    private static final String TAG = "PaymentAdapter";


    List<Integer> monthlyPaymentList = new ArrayList<>();

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // when there is not view holder, it creates a new view holder,
        // which later will be updated from the bound data.

        // Here the layout inflater takes the work list item layout file
        // and turns it into a view
        // later these views are used to draw on screen
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_payment,
                parent, false);

        // a new view holder is created based on the view of work list item.
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        // binds the new data to the view holder without creating new view holder to reduce cpu
        // as well as the memory usage.

        if (monthlyPaymentList.size() > 0) {
            holder.paymentListMonthNameTextView.setText(CustomUtils.getCurrentMonthName(position));
            holder.paymentListPaymentAmountTextView.setText(String.format("%s", monthlyPaymentList.get(position)));
        }
        String monthlyPaymentListString = monthlyPaymentList.toString();
        Log.d(TAG, "onBindViewHolder: " + monthlyPaymentListString);
    }

    @Override
    public int getItemCount() {
        return monthlyPaymentList.size();
    }

    public void setMonthlyPaymentList(List<Integer> integers) {
        this.monthlyPaymentList = integers;
        notifyDataSetChanged();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        final TextView paymentListMonthNameTextView;
        final TextView paymentListPaymentAmountTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            paymentListMonthNameTextView = itemView.findViewById(R.id.tv_payment_month_list_item_payment);
            paymentListPaymentAmountTextView = itemView.findViewById(R.id.tv_payment_amount_list_item_payment);
        }
    }
}

File: app/src/main/java/com/kabirnayeem99/paymentpaid/adapters/PaymentAdapter.kt
package com.kabirnayeem99.paymentpaid.adapters

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.AsyncListDiffer
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.RecyclerView
import com.kabirnayeem99.paymentpaid.R
import com.kabirnayeem99.paymentpaid.data.db.entities.Work
import com.kabirnayeem99.paymentpaid.utils.CustomUtils
import kotlinx.android.synthetic.main.list_item_payment.view.*
import java.util.*

class PaymentAdapter : RecyclerView.Adapter<PaymentAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    private var diffCallBack = object : DiffUtil.ItemCallback<Int>() {


        override fun areItemsTheSame(oldItem: Int, newItem: Int): Boolean {
            return oldItem == newItem
        }

        override fun areContentsTheSame(oldItem: Int, newItem: Int): Boolean {
            return oldItem == newItem
        }
    }

    var differ = AsyncListDiffer(this, diffCallBack)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item_payment,
                parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        if (differ.currentList.isNotEmpty()) {
            holder.itemView.apply {
                tvPaymentMonthListItemPayment.text = CustomUtils.getCurrentMonthName(position)
                tvPaymentAmountListItemPayment.text = differ.currentList[position].toString()
            }
            Log.d(TAG, "onBindViewHolder: ${differ.currentList.toString()}")
        }
    }


    override fun getItemCount(): Int {
        return differ.currentList.size
    }


    companion object {
        private const val TAG = "PaymentAdapter"
    }
}